src/opaque.rs

```diff
@@ -33,19 +36,48 @@ pub struct RegisterFirstMessage<Grp> {
 
 impl<Grp: Group> TryFrom<&[u8]> for RegisterFirstMessage<Grp> {
     type Error = ProtocolError;
-    fn try_from(first_message_bytes: &[u8]) -> Result<Self, Self::Error> {
+    fn try_from(input: &[u8]) -> Result<Self, Self::Error> {
+        if input[0] != REGISTRATION_REQUEST {
+            return Err(ProtocolError::VerificationError(
+                PakeError::SerializationError,
```

Comment:
what about using an `into()` or `?` on `SerializationError` directly?

Reply:
Can you clarify what you mean by this?